### PR TITLE
Improved ACA premium tax credit for California

### PR DESCRIPTION
**Pull request by martinholmer**
This pull request is the first step described in issue #3194.

### Summary
:sparkles::card_file_box::test_tube:

This pull request implements the logic and parameters for calculating the ACA premium tax credit for California residents, based on their income, age, and geographic rating area. It also adds unit and integration tests to verify the accuracy and consistency of the results. It introduces new parameter files for the ACA eligibility and phase-out rules, as well as new variable files for the ACA MAGI, the second-lowest-cost silver plan premium, and the ACA premium tax credit amount. It updates the changelog file to reflect the new feature and the minor version bump.

> _`aca_ptc_ca`_
> _New feature for tax credit_
> _Autumn of reform_

### Walkthrough
*  Add a new feature to improve the ACA premium tax credit for California residents ([link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4))
*  Define new parameters for the ACA eligibility and premium tax credit calculation, such as the income threshold, phase-out rate, maximum age, and geographic rating areas ([link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-72b74929b334a537e4980e938803ccb61997ec110427fa1fd799032415f8b195R1-R165), [link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-4b60c4b7bf8fa67ca772a97ad0247a637aa8ab56f448a78827c2b6616f02bde8R1-R9), [link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-959f88a54a42df26f0ddd6234473756236626108c08697a8012cb3fce30981e8R1-R9), [link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-9ba4bc059ab9738f5f4f8b03e2ad10d8a547c63df5329a63c162e95eaa1264fcR1-R9), [link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-8de838553d8ab85f701c57bc51f759a8222a303424dfdcfa7d163340ef1bc474R1-R9), [link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-60d070f305caf23b2f4a500ead731f8a7ef3581e1a5a9ac76efbd358a1ad0c85R1-R29), [link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-cf1d7cdbda3890f02e36d60e49511468ee9271181d8312db8f044f3ec2336bf8R1-R60))
*  Calculate the ACA-related modified AGI for a tax unit using the `medicaid_income` variable and the prior-year FPL ([link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-9f29c53b13dc10bb952bfe0018b8b9a33c4ada358768254820584697a33d7f7fR1-R9), [link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-8fae1de771ed9788a5e4edad8f0573fe8df9f2ced3553c360923e0640d79dd97R1-R22))
*  Determine the eligibility for the ACA premium tax credit based on the coverage and income status of each person in a tax unit ([link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-dd1b85053bad038b9aef12470ffe7d6cba826bd5b2f4888e953d9955df39bc4aR1-R85))
*  Determine the eligibility for Medicare based on the age of each person in a tax unit ([link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-782747665358ad50d81f7207b25c837df0c0c42974559253f4b90c2df618eed7R1-R6))
*  Assign an index to each child in a tax unit based on their age for the purpose of capping the number of children who pay an age-based premium ([link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-f9d4c13a3d2afaab0d44aba98b1c30f5cac8e69658c6df0413c7bd538456d54bR1-R14))
*  Trim the age of children and adults for the purpose of calculating the second-lowest-cost silver plan premium ([link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-bb33a537f4144f7d292267737929bc0c43bf49e0df7d35853885745a699fdd96R1-R6))
*  Determine the second-lowest-cost silver plan premium for a person in California based on their age and county or ZIP3 area using the `gra.yaml` parameter file ([link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-a0a3a8f12586d1fa0608b98c13e00c99abf691bf2877d05561cffc6044b1fc38R1-R9))
*  Calculate the ACA premium tax credit amount for California residents based on their income and second-lowest-cost silver plan premium ([link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-735602299e7679354399bcf02cc2dba752114ac7caf7acc741ad5c8559270902R1-R22))
*  Determine the applicable figure for the ACA premium tax credit based on the ACA MAGI percent of FPL using the `ptc_phase_out_rate.yaml` parameter file ([link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-c6e9738d15c9894961bdcfae2d747bb287d598a120bbdf1936156427caf327bdR1-R55))
*  Add unit tests for the new variables and parameters, and compare the results with the KFF ACA calculator for Los Angeles residents ([link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-0d2a5c25e2a3fe20ec601cb39254022c963a01c4fa86c07c742def45127cc0afR1-R21), [link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-dd1b85053bad038b9aef12470ffe7d6cba826bd5b2f4888e953d9955df39bc4aR1-R85), [link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-782747665358ad50d81f7207b25c837df0c0c42974559253f4b90c2df618eed7R1-R6), [link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-735602299e7679354399bcf02cc2dba752114ac7caf7acc741ad5c8559270902R1-R22), [link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-668b6ace1567ac6a77ca6017dc2f1c34a2e052b4180b0ec2bd8b699e2d658ed8R1-R198), [link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-c6e9738d15c9894961bdcfae2d747bb287d598a120bbdf1936156427caf327bdR1-R55), [link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-bb33a537f4144f7d292267737929bc0c43bf49e0df7d35853885745a699fdd96R1-R6), [link](https://github.com/PolicyEngine/policyengine-us/pull/3261/files?diff=unified&w=0#diff-a0a3a8f12586d1fa0608b98c13e00c99abf691bf2877d05561cffc6044b1fc38R1-R9))

### CODE REVIEW

File: policyengine_us/data/parameters/gov/aca/ca/gra.yaml [hunk 1]
```diff
@@ -0,0 +1,165 @@
+description: ACA geographic rating areas for California.
+metadata:
+  reference:
```
Comment (MaxGhenis) [inline policyengine_us/data/parameters/gov/aca/ca/gra.yaml:3]: Could you add a breakdown metadata field? Here and elsewhere

File: policyengine_us/data/parameters/gov/aca/ca/gra.yaml [hunk 2]
```diff
@@ -0,0 +1,165 @@
+description: ACA geographic rating areas for California.
```
Comment (MaxGhenis) [inline policyengine_us/data/parameters/gov/aca/ca/gra.yaml:1]:
```suggestion
description: California assigns counties to these Affordable Care Act geographic rating areas.
```
Example for other descriptions

File: policyengine_us/data/parameters/gov/aca/ca/gra.yaml (hunk 1 above)
Comment (MaxGhenis) [inline policyengine_us/data/parameters/gov/aca/ca/gra.yaml:3]:
```suggestion
  label: California ACA geographic rating areas
  reference:
  
```
Please add labels to all parameters

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/data/parameters/gov/aca/ca/gra.yaml (hunk 1 above)
Comment (martinholmer) [reply policyengine_us/data/parameters/gov/aca/ca/gra.yaml:3]: What is a "breakdown metadata field" and what kind of parameters should have one?

File: policyengine_us/data/parameters/gov/aca/ca/gra.yaml (hunk 2 above)
Comment (martinholmer) [reply policyengine_us/data/parameters/gov/aca/ca/gra.yaml:1]: The problem with your suggestion
```
California assigns counties to these Affordable Care Act geographic rating areas.
```
is that it is incorrect.

File: policyengine_us/variables/gov/aca/eligible/aca_magi_percent.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        "Documentation on truncation of percent in the following reference:"
+        "  title: 2022 IRS Form 8962 instructions, Line 5 Worksheet 2"
+        "  href: https://www.irs.gov/pub/irs-pdf/i8962.pdf#page=8"
+    )
+    definition_period = YEAR
+
+    def formula(tax_unit, period, parameters):
+        magi = max_(0, tax_unit("aca_magi", period))
+        fpg = tax_unit("tax_unit_fpg", period.last_year)
+        return np.floor(100.0 * magi / fpg)
```
[the remark this answers is not in the document]
Comment (martinholmer) [reply policyengine_us/variables/gov/aca/eligible/aca_magi_percent.py:22]: Internal consistency is one goal, but representing policy parameters the way the legislation and policy analysts describe them is, in my mind, an even more important goal.   If you want internal consistency, I suggest you change all the Medicaid-related policy parameter to percentage terms (e.g., 1.33 --> 133).

File: policyengine_us/variables/gov/aca/eligible/is_medicare_eligible.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+
+class is_medicare_eligible(Variable):
+    value_type = bool
+    entity = Person
+    label = "Person is eligible for Medicare"
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        return person("age", period) >= 65
```
[the remark this answers is not in the document]
Comment (martinholmer) [reply policyengine_us/variables/gov/aca/eligible/is_medicare_eligible.py:11]: This is the Medicare eligibility age.   Is PolicyEngine-US really planning to model the Medicare program?   I have never heard or seen any plans for that to happen.  If it is added as a parameter, you'll just have to add something to make it non-changeable by your users.   And, as far as I can see, Nikhil did not specify this as a parameter in his original ACA development.   Why, after almost a year, is it now important to make this an explicit parameter?   I will be happy to add a descriptive constant to the `is_medicaid_eligible` formula that clarifies the meaning of 65, which is I thought was pretty obvious in the current version.

File: policyengine_us/data/parameters/gov/aca/ca/gra.yaml (hunk 1 above)
Comment (martinholmer) [reply policyengine_us/data/parameters/gov/aca/ca/gra.yaml:3]: I was just following the example set by Nikhil in his original ACA implementation, in which parameters have a `name:`.   If that is wrong, why did he do it that way?   I don't see anywhere in the GitHub repo where the "rules" you are describing are specified.   Have I missed a link to that information?

File: policyengine_us/variables/gov/aca/ptc/aca_ptc_phase_out_rate.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    value_type = float
+    entity = TaxUnit
+    label = "ACA PTC phase-out rate (i.e., IRS Form 8962 'applicable figure')"
+    unit = "/1"
+    definition_period = YEAR
+    reference = "https://www.law.cornell.edu/uscode/text/26/36B#b_3_A"
+
+    def formula(tax_unit, period, parameters):
+        magi_pct = tax_unit("aca_magi_percent", period)
+        porate = parameters(period).gov.aca.ptc_phase_out_rate
```
[the remark this answers is not in the document]
Comment (martinholmer) [reply policyengine_us/variables/gov/aca/ptc/aca_ptc_phase_out_rate.py]: DONE

File: policyengine_us/variables/gov/aca/eligible/aca_magi_percent.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+        "  href: https://www.irs.gov/pub/irs-pdf/i8962.pdf#page=7"
+        "Documentation on truncation of percent in the following reference:"
+        "  title: 2022 IRS Form 8962 instructions, Line 5 Worksheet 2"
+        "  href: https://www.irs.gov/pub/irs-pdf/i8962.pdf#page=8"
+    )
+    definition_period = YEAR
+
+    def formula(tax_unit, period, parameters):
+        magi = max_(0, tax_unit("aca_magi", period))
+        fpg = tax_unit("tax_unit_fpg", period.last_year)
```
[the remark this answers is not in the document]
Comment (martinholmer) [reply policyengine_us/variables/gov/aca/eligible/aca_magi_percent.py]: This suggested comment is redundant given that the documentation already discusses this and provides a link to the relevant IRS instructions.

File: policyengine_us/variables/gov/aca/eligible/aca_child_index.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+class aca_child_index(Variable):
+    value_type = int
+    entity = Person
+    label = "Index of child in tax unit"
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        age = person("age", period)
+        is_child = age <= parameters(period).gov.aca.max_child_age
+        child_rank = person.get_rank(person.tax_unit, -age, condition=is_child)
```
[the remark this answers is not in the document]
Comment (martinholmer) [reply policyengine_us/variables/gov/aca/eligible/aca_child_index.py]: DONE

File: policyengine_us/data/parameters/gov/aca/ca/gra.yaml (hunk 1 above)
Comment (martinholmer) [reply policyengine_us/data/parameters/gov/aca/ca/gra.yaml:4]: Revised description and added label to both the `data/parameters/aca/ca/gra.yaml` and `data/parameters/aca/ca/slspc.yaml` parameter files.

I think all the other parameter files have a label.

Comment (martinholmer) [reply policyengine_us/data/parameters/gov/aca/ca/gra.yaml:4]: DONE

File: policyengine_us/data/parameters/gov/aca/ca/gra.yaml (hunk 2 above)
Comment (martinholmer) [reply policyengine_us/data/parameters/gov/aca/ca/gra.yaml]: Have revised the description for both the `gra.yaml` and `slspc.yaml` parameter files.

Comment (martinholmer) [reply policyengine_us/data/parameters/gov/aca/ca/gra.yaml]: DONE

File: policyengine_us/parameters/gov/aca/max_adult_age.yaml [hunk 8]
```diff
@@ -0,0 +1,9 @@
+description: Maximum age for adults under the ACA.
+metadata:
+  period: year
+  label: ACA maximum adult age
+  reference:
+    - title:
```
[the remark this answers is not in the document]
Comment (martinholmer) [reply policyengine_us/parameters/gov/aca/max_adult_age.yaml]: DONE

File: policyengine_us/variables/gov/aca/eligible/is_medicare_eligible.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (martinholmer) [reply policyengine_us/variables/gov/aca/eligible/is_medicare_eligible.py]: I have added documentation on the Medicare eligibility age to the `is_medicare_eligible` variable metadata.

File: policyengine_us/data/parameters/gov/aca/ca/gra.yaml (hunk 1 above)
Comment (martinholmer) [reply policyengine_us/data/parameters/gov/aca/ca/gra.yaml:6]: @MaxGhenis, you haven't provided any information about how to specify a "breakdown metadata field" and I can't find any written developer information about it on the PolicyEngine-US GitHub site, so I was forced to experiment.  

The first thing I did was to consult Nikhil's `parameters/gov/hhs/.../medicaid_rating_area.yaml` file`.  However, that parameter file has **no breakdown metadata** as can be seen from this screenshot:

Screenshot 2023-11-14 at 9 31 00 AM


When I add breakdown metadata to the new `parameters/gov/aca/ca/gra.yaml` file, I get an error.  Here is what I added:
```
  breakdown:
    - county_fips
```
And here is the error I get:

Screenshot 2023-11-14 at 9 38 11 AM


The first thing to note is that the error is in the `policyengine-core` code.   
The second thing I noticed is that it appears as if the `policyengine-core` homogenize_parameters logic thinks `county_fips` is an enumeration (rather than an integer) variable.  Why would it think that?  Or maybe my interpretation is incorrect.

In conclusion, I cannot get your suggestion to add a "breakdown metadata field" to the `gra.yaml` parameter file to work.   And I'm puzzled about why you would make such a suggestion given that Nikhil's rating area parameter file does not contain any breakdown metadata.  (And there plenty of other parameter files in the code that appear to have missing breakdown metadata.).  I have no idea how to proceed.

File: policyengine_us/variables/gov/aca/eligible/aca_magi_percent.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (martinholmer) [reply policyengine_us/variables/gov/aca/eligible/aca_magi_percent.py]: @MaxGhenis,  at your suggestion I checked the SNAP income eligibility parameters, which are defined in the `.../snap/income/limit/gross.yaml` and `.../snap/income/limit/gross.yaml` files.   Those two parameter files provide the same [reference to the US tax code](https://www.law.cornell.edu/cfr/text/7/273.9#a_2), which says (in part):

Screenshot 2023-11-14 at 11 41 54 AM



So, the legislation characterizes the gross income limit as **130%** of the FPL, not as 1.3 times the FPL (as in the parameter file).

So, in conclusion, it seems to me that the SNAP income-to-FPL parameters in the Policyengine-US source code are at variance with the legislative language.

[the remark this answers is not in the document]
Comment (MaxGhenis) [reply policyengine_us/variables/gov/aca/eligible/aca_magi_percent.py]: `x%` is just a way to display `x/100`. Most percentages in the tax code, including those you've implemented, are written as `x%` in the legal code, yet we express them as decimals. We denote them with the `unit: /1` metadata field, which instructs our app to display them as percentages.

As an example, the [IRC](https://www.law.cornell.edu/uscode/text/26/3101#a) defines the employee-side OASDI payroll tax as:
>In addition to other taxes, there is hereby imposed on the income of every individual a tax equal to **6.2 percent** of the [wages](https://www.law.cornell.edu/definitions/uscode.php?width=840&height=800&iframe=true&def_id=26-USC-112890955-191573808&term_occur=999&term_src=title:26:subtitle:C:chapter:21:subchapter:A:section:3101) (as defined in [section 3121(a)](https://www.law.cornell.edu/uscode/text/26/3121#a)) received by the individual with respect to[ employment ](https://www.law.cornell.edu/definitions/uscode.php?width=840&height=800&iframe=true&def_id=26-USC-168272876-1501873671&term_occur=999&term_src=)(as defined in section 3121(b)).

We code this as:
https://github.com/PolicyEngine/policyengine-us/blob/36c053c7a3c452bbbfaa8cccbfe6349ed053acfa/policyengine_us/parameters/gov/irs/payroll/social_security/rate/employee.yaml#L1-L10

And [display it as](https://policyengine.org/us/policy?focus=gov.irs.payroll.social_security.rate.employee):
image

I don't see why we would treat ACA differently.

File: policyengine_us/variables/gov/aca/eligibility/is_medicare_eligible.py [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    definition_period = YEAR
+    documentation = (
+        "CMS, Original Medicare (Part A and B) Eligibility and Enrollment"
+        "https://www.cms.gov/medicare/enrollment-renewal/health-plans/original-part-a-b"
+        "Above link includes the following text:"
+        "  Part A coverage begins the month the individual turns age 65"
+    )
+
+    def formula(person, period, parameters):
+        MEDICARE_ELIGILITY_AGE = 65
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/aca/eligibility/is_medicare_eligible.py:17]: can you make this a policy parameter? here's a source https://www.law.cornell.edu/uscode/text/42/426#a_1

File: policyengine_us/variables/gov/aca/slspc/person_aca_slspc_ca.py [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+                / "ca"
+            )
+            aca_params = homogenize_parameter_structures(
+                aca_params,
+                person.simulation.tax_benefit_system.variables,
+            )
+            ptree.gov.aca.add_child("ca_", aca_params)
+        # specify ACA geographic rating area
+        aca_gra = ptree.gov.aca.ca_.gra(period)
+        in_state = person.household("state_fips", period) == 6
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/aca/slspc/person_aca_slspc_ca.py:34]:
```suggestion
        CA_FIPS = 6
        in_state = person.household("state_fips", period) == CA_FIPS
```

File: policyengine_us/variables/gov/aca/slspc/person_aca_slspc_ca.py [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+        cofips = person.household("county_fips", period)
+        in_la_county = cofips == 37
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/aca/slspc/person_aca_slspc_ca.py:36]:
```suggestion
        LA_COUNTY_FIPS = 37
        in_la_county = cofips == LA_COUNTY_FIPS
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/hhs/medicare/eligibility/min_age.yaml [hunk 12]
```diff
@@ -0,0 +1,9 @@
+description: Minimum age for age-based Medicare eligibility.
+metadata:
+  period: year
+  label: Minimum age for age-based Medicare eligibility
+  reference:
+    - title: 42 U.S. Code § 426
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/hhs/medicare/eligibility/min_age.yaml:6]:
```suggestion
    - title: 42 U.S. Code § 426(a)(1)
```

File: policyengine_us/parameters/gov/hhs/medicare/eligibility/min_age.yaml [hunk 13]
```diff
@@ -0,0 +1,9 @@
...
+      href: https://www.law.cornell.edu/uscode/text/42/426
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/hhs/medicare/eligibility/min_age.yaml:7]:
```suggestion
      href: https://www.law.cornell.edu/uscode/text/42/426#a_1
```

File: policyengine_us/parameters/gov/hhs/medicare/eligibility/min_months_receiving_social_security_disability.yaml [hunk 14]
```diff
@@ -0,0 +1,9 @@
+description: Minimum number of months of receiving social security disability for Medicare eligibility.
+metadata:
+  period: year
+  label: Minimum number of months of receiving social security disability for Medicare eligibility
+  reference:
+    - title: 42 U.S. Code § 426
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/hhs/medicare/eligibility/min_months_receiving_social_security_disability.yaml:6]:
```suggestion
    - title: 42 U.S. Code § 426(b)(2)(A)
```

File: policyengine_us/parameters/gov/hhs/medicare/eligibility/min_months_receiving_social_security_disability.yaml [hunk 15]
```diff
@@ -0,0 +1,9 @@
...
+      href: https://www.law.cornell.edu/uscode/text/42/426
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/hhs/medicare/eligibility/min_months_receiving_social_security_disability.yaml:7]:
```suggestion
      href: https://www.law.cornell.edu/uscode/text/42/426#b_2_A
```

File: policyengine_us/variables/gov/aca/eligibility/is_aca_ptc_eligible.py [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+        # determine coverage eligibility for ACA plan
+        medicaid_coverage = person("is_medicaid_eligible", period)
+        eshi_coverage = person("is_aca_eshi_eligible", period)
+        medicare_coverage = person("is_medicare_eligible", period)
+        is_coverage_eligible = (
+            ~medicaid_coverage & ~eshi_coverage & ~medicare_coverage
+        )
+
+        # determine income eligibility for ACA PTC
+        aca = parameters(period).gov.aca
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/aca/eligibility/is_aca_ptc_eligible.py:22]:
```suggestion
        p = parameters(period).gov.aca
```

File: policyengine_us/variables/gov/aca/eligibility/is_aca_ptc_eligible.py [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+        magi_frac = person.tax_unit("aca_magi_fraction", period)
+        is_income_eligible = aca.ptc_income_eligibility.calc(magi_frac)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/aca/eligibility/is_aca_ptc_eligible.py:24]:
```suggestion
        is_income_eligible = p.ptc_income_eligibility.calc(magi_frac)
```

File: policyengine_us/variables/gov/aca/eligibility/is_aca_ptc_eligible.py [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+
+        # determine which people pay an age-based ACA plan premium
+        is_aca_adult = person("age", period) > aca.max_child_age
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/aca/eligibility/is_aca_ptc_eligible.py:27]:
```suggestion
        is_aca_adult = person("age", period) > p.max_child_age
```

File: policyengine_us/variables/gov/aca/eligibility/is_aca_ptc_eligible.py [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+        child_pays = person("aca_child_index", period) <= aca.max_child_count
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/aca/eligibility/is_aca_ptc_eligible.py:28]:
```suggestion
        child_pays = person("aca_child_index", period) <= p.max_child_count
```

File: policyengine_us/parameters/gov/aca/ptc_phase_out_rate.yaml [hunk 20]
```diff
@@ -0,0 +1,60 @@
+description: Rate of ACA PTC phase out by ACA MAGI percent of FPL.
+metadata:
+  type: single_amount
+  threshold_unit: /1
+  rate_unit: /1
+  name: aca_ptc_phase_out_rate
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/aca/ptc_phase_out_rate.yaml:6]: We deprecated this field
```suggestion
```

File: policyengine_us/parameters/gov/aca/ptc_phase_out_rate.yaml [hunk 21]
```diff
@@ -0,0 +1,60 @@
...
+  reference:
+    - title: 26 U.S. Code § 36B - Refundable credit for coverage under a qualified health plan
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/aca/ptc_phase_out_rate.yaml:8]:
```suggestion
    - title: 26 U.S. Code § 36B(b)(3)(A) - Refundable credit for coverage under a qualified health plan
      # (i) in general; (iii) for 2021-2025.
```

File: policyengine_us/parameters/gov/aca/ptc_phase_out_rate.yaml [hunk 22]
```diff
@@ -0,0 +1,60 @@
...
+      href: https://www.law.cornell.edu/uscode/text/26/36B
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/aca/ptc_phase_out_rate.yaml:9]:
```suggestion
      href: https://www.law.cornell.edu/uscode/text/26/36B#b_3_A
```

File: policyengine_us/parameters/gov/aca/ptc_income_eligibility.yaml [hunk 23]
```diff
@@ -0,0 +1,29 @@
+description: Eligibility for ACA Premium Tax Credit by percent of modified AGI to federal poverty line.
+metadata:
+  type: single_amount
+  period: year
+  threshold_unit: /1
+  amount_unit: bool
+  label: ACA PTC income eligibility conditions
+  reference:
+    - title: 26 U.S. Code § 36B - Refundable credit for coverage under a qualified health plan
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/aca/ptc_income_eligibility.yaml:9]:
```suggestion
    - title: 26 U.S. Code § 36B(c)(1) - Refundable credit for coverage under a qualified health plan
      # (A) in general; (E) for 2021-2025.
```

File: policyengine_us/parameters/gov/aca/ptc_income_eligibility.yaml [hunk 24]
```diff
@@ -0,0 +1,29 @@
...
+      href: https://www.law.cornell.edu/uscode/text/26/36B
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/aca/ptc_income_eligibility.yaml:10]:
```suggestion
      href: https://www.law.cornell.edu/uscode/text/26/36B#c_1
```

File: policyengine_us/parameters/gov/aca/max_adult_age.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/aca/max_adult_age.yaml:1]: Do we need this given the parameter for 65 Medicare age eligibility? If it legally derives from Medicare we can omit it.

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/aca/max_adult_age.yaml
Comment (MaxGhenis) [reply policyengine_us/parameters/gov/aca/max_adult_age.yaml:1]: Just reupping this question

File: policyengine_us/variables/gov/hhs/medicaid/eligibility/is_medicaid_eligible.py [hunk 25]
```diff
@@ -6,8 +6,14 @@ class is_medicaid_eligible(Variable):
...
+        "https://www.law.cornell.edu/uscode/text/42/1396a#a_10"
+        "https://www.kff.org/racial-equity-and-health-policy/fact-sheet/key-facts-on-health-coverage-of-immigrants"
+    )
 
     def formula(person, period, parameters):
         category = person("medicaid_category", period)
-        return category != category.possible_values.NONE
+        categorically_eligible = category != category.possible_values.NONE
+        istatus = person("immigration_status", period)
+        undocumented = istatus == istatus.possible_values.UNDOCUMENTED
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/hhs/medicaid/eligibility/is_medicaid_eligible.py:18]: For posterity, we're merging this for now but https://github.com/PolicyEngine/policyengine-us/issues/3324

**Review by MaxGhenis — APPROVED** [review]
